core/platform/storage/cloud_storage_emulator.py

```diff
@@ -0,0 +1,242 @@
+# coding: utf-8
+#
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""An emulator that mocks the core.platform.storage API."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals # pylint: disable=import-only-modules
+
+import mimetypes
+
+import feconf
+import python_utils
+
+import redis
+
+REDIS_CLIENT = redis.StrictRedis(
+    host=feconf.REDISHOST,
+    port=feconf.REDISPORT,
+    db=feconf.STORAGE_EMULATOR_REDIS_DB_INDEX
+)
+
+
+class Blob(python_utils.OBJECT):
+    """Object for storing the file data."""
+
+    def __init__(self, name, data, content_type):
+        """Initialize blob.
+
+        Args:
+            name: str. The name of the blob.
+            data: str|bytes. The data of the blob. If the data are string,
+                they are encoded to bytes.
```

Comment:
Maybe add at the end: Note that data is *always* retrieved from cloud storage as bytes.

Reply:
Done.